.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,60 @@
+name: Tests
+on:
+  push:
+    branches:
+      - master
+  # Run tests against the forked branch, but
+  # do not allow access to secrets
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories
+  pull_request:
+
+env:
+  SDK_VERSION: "7.0.x"
+  TEST_ARGS: >
+    --configuration Release
+    --collect:"XPlat Code Coverage"
+    --settings tests/coverletArgs.runsettings
+    --verbosity minimal
+
+
+jobs:
+  run-tests:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+
+    runs-on: "${{ matrix.os }}"
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.SDK_VERSION }}
+
+      - name: Run DotNet CLI Tests
+        run: >
+          find tests -type f -iname "*.csproj" -print0 | xargs -0 -I{} -t
+          dotnet test '{}' ${{ env.TEST_ARGS }}
+        if: matrix.os != 'windows-latest'
+
+      - name: Run DotNet CLI Tests
+        run: >
+          Get-ChildItem -Recurse -Filter *.csproj -Path tests |
```

Comment:
With the pattern above we won't need separate tasks

Reply:
I couldn't get this to pick up all the tests when I tested it locally, and there doesn't seem to be anything on the [docs about recursive](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-test#arguments)

Attempting to run it gives:
```
  dotnet test 'tests/**/*.csproj' --configuration Release --collect:"XPlat Code Coverage" --settings tests/coverletArgs.runsettings --verbosity minimal
  shell: /usr/bin/bash -e {0}
  env:
    SDK_VERSION: 7.0.x
    TEST_ARGS: --configuration Release --collect:"XPlat Code Coverage" --settings tests/coverletArgs.runsettings --verbosity minimal
  
    DOTNET_ROOT: /usr/share/dotnet
  
MSBUILD : error MSB1009: Project file does not exist.
Switch: tests/**/*.csproj
```

And it looks like it's not planned: https://github.com/dotnet/sdk/issues/30655

I suspect the Azure equivalent is effectively a wrapper round a glob and something similar to this. I'm happy to take suggestions what we could do instead as something cross-platform would be a big plus in my books too

Reply:
Unfortunate. We only have test projects in the tests folder. We could just use `dotnet test Jellyfin.sln`

Reply:
:+1: I'd be happy with that
Let me rejig this PR and I can add an extra check too that all of the `*Tests.csproj` are named in the `.sln` too, so we don't have any slip through the cracks...